template-for-repository/proofs/Makefile.common

```diff
@@ -305,23 +305,26 @@ PROOF_SOURCES ?=
 # report, making the proof run appear to "hang".
 CBMC_TIMEOUT ?= 21600
 
-# Proof writers could add function contracts or loop invariants in
-# CBMC proofs. They should follow three distinct processes:
-# 1. Check whether a function contract is sound, where CHECK_FUNCTION_CONTRACTS
-#    contains the list of functions with contracts to be checked.
-# 2. Replace function calls with their correspondent function contracts,
-#    where USE_FUNCTION_CONTRACTS contains the list of functions with
-#    contracts to be replaced. One must check separately whether a function
-#    contract is sound before replacing it.
-# 3. Apply loop invariants during verification, where APPLY_LOOP_INVARIANTS
-#    contains the list of functions with loop invariants to be applied.
-#    This option is currently unavailable.
+# Proof writers could add function contracts or loop contracts in their source
+# code. These contracts are ignored by default. They may be enabled using three
+# distinct flags:
+# 1. To check whether one or more function contracts are sound with respect to
+#    the function implementation, CHECK_FUNCTION_CONTRACTS should be a list of
+#    function names.
+# 2. To replace calls to certain functions with their correspondent function
+#    contracts, USE_FUNCTION_CONTRACTS should be a list of function names.
+#    One must check separately whether a function contract is sound before
+#    replacing it in calling contexts.
+# 3. To use loop contracts (invariants and variants) for unbounded verification,
+#    USE_LOOP_CONTRACTS should be set to "TRUE".
 CHECK_FUNCTION_CONTRACTS ?= ""
 CBMC_CHECK_FUNCTION_CONTRACTS := $(patsubst %,--enforce-contract %, $(CHECK_FUNCTION_CONTRACTS))
 
 USE_FUNCTION_CONTRACTS ?= ""
 CBMC_USE_FUNCTION_CONTRACTS := $(patsubst %,--replace-call-with-contract %, $(USE_FUNCTION_CONTRACTS))
 
+USE_LOOP_CONTRACTS ?= FALSE
```

Comment:
Let's apply the new macro from https://github.com/diffblue/cbmc/pull/6184. We can only update starter kits in projects once CBMC's PR get merge and this will only break builds that tries to use it without support.

Reply:
I don't think I understand this comment about macro. What do CBMC macros have to do with Makefiles? ...

Reply:
> We can only update starter kits in projects once CBMC's PR get merge and this will only break builds that tries to use it without support.

But builds shouldn't be affected if projects don't push the starter kit pointer forward. Function contract support was added in a previous commit and most repos have already forwarded the pointer.

I wanted to merge this early so that the interns wouldn't have to maintain a separate starter-kit branch _and_ a separate project branch and so on. What do you suggest?